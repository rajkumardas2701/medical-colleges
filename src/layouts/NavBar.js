import '../styles/NavBar.css';

const NavBar = () => (
  <navbar>
    <a href="/" className="logoAnc">
      <h1>IndianMedicalColleges</h1>
    </a>
    <a href="/" className="btnAnc">
      <button type="button">
        Home
      </button>
    </a>
  </navbar>
);

export default NavBar;
