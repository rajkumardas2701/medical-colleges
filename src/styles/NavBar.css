navbar {
  display: flex;
  flex-direction: row;
  background: rgb(41, 29, 29);
  height: 50px;
  align-items: center;
  justify-content: space-between;
  position: fixed;
  left: 0;
  top: 0;
  width: 100%;
  justify-items: center;
  z-index: 90;
}

h1 {
  color: whitesmoke;
  width: 20%;
  text-align: center;
  font-family: 'Courier New', Courier, monospace;
  margin: 6px;
}

.logoAnc {
  margin: 0 100px;
  width: 20%;
  text-decoration: none;
  height: 50px;
}

.btnAnc > button {
  color: rgba(8, 7, 7, 0.856);
  height: 40px;
  font-size: 23px;
  font-family: 'Courier New', Courier, monospace;
  font-weight: bolder;
  border-radius: 7px;
  border-style: none;
}

.btnAnc,
button {
  margin-right: 150px;
  width: 100px;
  height: 40px;
}

button:hover {
  cursor: pointer;
}
