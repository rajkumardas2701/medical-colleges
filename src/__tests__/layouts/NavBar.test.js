import React from 'react';
import renderer from 'react-test-renderer';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NavBar from '../../layouts/NavBar';

afterEach(cleanup);

it('renders Navbar component correctly', () => {
  const nav = renderer.create(<NavBar />).toJSON();
  expect(nav).toMatchSnapshot();
});

it('should display the heading', () => {
  const { getByTestId } = render(<NavBar />);
  expect(getByTestId('navbar-heading')).toHaveTextContent('IndianMedicalColleges');
});
