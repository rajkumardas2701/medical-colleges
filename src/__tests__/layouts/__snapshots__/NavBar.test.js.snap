// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar component correctly 1`] = `
<div
  className="navbar"
>
  <a
    className="logoAnc"
    href="/"
  >
    <h1
      data-testid="navbar-heading"
    >
      IndianMedicalColleges
    </h1>
  </a>
  <a
    className="btnAnc"
    href="/"
  >
    <button
      type="button"
    >
      Home
    </button>
  </a>
</div>
`;
